pkg/occlient/occlient.go

```diff
@@ -631,6 +631,34 @@ func (c *Client) NewAppS2I(name string, builderImage string, gitUrl string, labe
 
 }
 
+func (c *Client) UpdateComponent(componentName string, projectName string, gitUrl string, annotations map[string]string) error {
```

Comment:
It might be better to call this function something like `UpdateBuildConfig`.  and change `componentName` to `buildConfigName`. OcClient shouldn't have any references to components or applications. It should be purely about openShift objects.